caravel/controllers/moderation.py

```diff
@@ -58,4 +85,42 @@ def moderation_view():
 
     return render_template("moderation/view.html",
                            inquiries=inquiries,
-                           listings=listings)
+                           listings=listings,
+                           config=config)
+
+
+@app.route("/_internal/automod")
+def automod():
+    if not users.is_current_user_admin():
+        return "???"
+
+    pending = itertools.chain(
+        model.UnapprovedListing().query(),
+        model.UnapprovedInquiry().query(),
+    )
+
+    config = ModerationConfig.get()
+    if not config.enabled:
+        return ""
+
+    for to_moderate in pending:
+        if to_moderate.age < datetime.timedelta(minutes=config.min_delay):
+            continue
+
+        values = dict(to_moderate.to_dict())
+        values["principal"] = values["principal"].email
+        allowed = True
+
+        for field in values.values():
+            for item in config.blacklist:
```

Comment:
Maybe put a note here: https://github.com/uchicago-sg/caravel/pull/239/files#diff-ad46fd1aeff1d04636c82ea26be28274R24

PS: is there a better way of referencing code numbers in comments?